docs/references/architecture/adr-113-mempool-lanes.md

```diff
@@ -0,0 +1,312 @@
+      so definitely we don't want 1
+  * 2. vs. 3.
+    * If we can change lane info via `ConsensusParams`
+      * CometBFT's mempool needs logic for changing lanes dynamically (complex, not really appealing for MVP)
+      * The process of updating lane info very complex and cumbersome:
+        * to update lanes you'd need to pass _two_ governance proposals
+          * upgrade the app, because the lane classification logic (so, the app's code) needs to know the lane config beforehand
+          * then upgrade the lanes via `ConsensusParams`
+        * also, not clear in which order: community should be careful not to break performance between the passing of both proposals
+        * the `gov` module may allow the two things to be shipped in the same gov proposal,
+          but, if we need to do it that way, what's the point in having lanes in `ConsensusParams` ?
+  * Conclusion: 3.
+    * lane info is "hardcoded" in the app's logic
+    * lane info is passed to CometBFT during handshake
+  * Advantages:
+    * Simple lane update cycle: one software update gov proposal
+    * CometBFT doesn't need to deal with dynamic lane changes: it just needs to set up lanes when starting up (whether afresh, or recovery)
+* What does lane info (passed to CometBFT) look like?
+  * Current state  of discussions.
+    * Draft of data structure
+        ```protobuf
+        message LaneInfo {
+          repeated Lane lanes;
+        }
+        message Lane {
+          byte id;
+          string name;
+        }
+        ```
+  * The `Lane` list MUST NOT have duplicate lane IDs
+  * The order of the `Lane` elements in the `lanes` field defines their priority
+  * Lane ID 0 is the native lane.
+    * It MAY be present in the list
+    * If it is absent, it is equivalent to having it as the last element (lowest priority)
+  * Channel ID is a byte
+    * Current channel distribution (among reactors) goes up to `0x61`
+    * Proposal: reserve for mempool lanes channel ID `0x80` and all channels above (so, all channels whose MSB is 1)
+      * max of 128 lanes. Big enough?
+    * currently, mempool is p2p channel ID is `0x30`, which would be a special case: native lane.
+  * How to deal with nodes that are late? So, lane info at mempool level (and thus p2p channels to establish) does not match.
+    * Two cases
+    * A node that is up to date and falls behind (e.g., network instability, etc.)
+      * Not a problem. For lanes to change we **require** a coordinated update.
+        * Reason: if we allow upgrade changing lane info **not** to be a coordinated upgrage,
+          then we won't have consistent lane info across nodes (each node would be running a particular version of the software)
+    * A node that is blocksyncing from far away in the past (software upgrades in the middle)
+      * Normal channels (including `0x30`): same as before
+      * Lane channels (>=`0x80`), not declared until `SwitchToConsensus`
```

Comment:
```suggestion
      * Lane channels (>=`0x80`), not declared. The node can't send/receive mempool-related info. Not a problem, since it's blocksyncing
```